test/phases/mystery-encounter-phase.test.ts

```diff
@@ -75,14 +75,14 @@ describe("Mystery Encounter Phases", () => {
       ]);
 
       game.onNextPrompt("MysteryEncounterPhase", UiMode.MESSAGE, () => {
-        const handler = game.scene.ui.getHandler() as MessageUiHandler;
+        const handler = game.scene.ui.getCurrentHandler() as MessageUiHandler;
```

Comment:
NIT: Use generic param `<T>`

```suggestion
        const handler = game.scene.ui.getCurrentHandler<MessageUiHandler>();
```